[Test][JITLink] Correctly generate the R_X86_64_PC8 relocation.

higuoxing opened the pull request:
Previously, ELF_R_X86_64_PC8.s doesn't produce the R_X86_64_PC8 relocation. This patch helps fix it.

Before:

```
llvm-mc -triple=x86_64-unknown-linux -position-independent -filetype=obj -o - ELF_R_X86_64_PC8.s | llvm-readobj -r -

File: <stdin>
Format: elf64-x86-64
Arch: x86_64
AddressSize: 64bit
LoadName: <Not found>
Relocations [
]
```

After:

```
llvm-mc -triple=x86_64-unknown-linux -position-independent -filetype=obj -o - ELF_R_X86_64_PC8.s | llvm-readobj -r -

File: <stdin>
Format: elf64-x86-64
Arch: x86_64
AddressSize: 64bit
LoadName: <Not found>
Relocations [
  Section (3) .rela.text {
    0x4 R_X86_64_PC8 main 0xFFFFFFFFFFFFFFFF
  }
]
```

maksfb commented on llvm/test/ExecutionEngine/JITLink/x86-64/ELF_R_X86_64_PC8.s:
```diff
@@ -10,10 +10,4 @@
 	retq
-	.size	main, .-main
-
-	.type	P,@object
```
Thanks for spotting the issue and providing the fix. I'm not sure why `loope` would make `llvm-mc` emit relocation and `.byte main-.` would not. I suggest not to rely on the `loope` behavior and issue `.byte main-.` in a different section, e.g. `.rodata`. I.e. if you insert `.rodata` before `.globl P`, it will force `llvm-mc` to always emit a relocation.

higuoxing replied:
Thanks for the suggestion! I fixed it by emitting a byte `main - .` to the `.rodata` section.

higuoxing committed:
[JITLink] Correctly generate the R_X86_64_PC8 relocation.

Previously, ELF_R_X86_64_PC8.s doesn't produce the R_X86_64_PC8
relocation. This patch helps fix it by emitting a byte `main - .` to the
.rodata section.

Before:

```
llvm-mc -triple=x86_64-unknown-linux -position-independent -filetype=obj -o - ELF_R_X86_64_PC8.s | llvm-readobj -r -

File: <stdin>
Format: elf64-x86-64
Arch: x86_64
AddressSize: 64bit
LoadName: <Not found>
Relocations [
]
```

After:

```
llvm-mc -triple=x86_64-unknown-linux -position-independent -filetype=obj -o - ELF_R_X86_64_PC8.s | llvm-readobj -r -

File: <stdin>
Format: elf64-x86-64
Arch: x86_64
AddressSize: 64bit
LoadName: <Not found>
Relocations [
  Section (4) .rela.rodata {
    0x0 R_X86_64_PC8 main 0x0
  }
]
```

lhames reviewed (approved):
LGTM. Thanks @higuoxing and @maksfb!
